From patchwork Fri Jun 18 18:12:52 2021
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Alex Bee <knaerzche@gmail.com>
X-Patchwork-Id: 12331871
Return-Path: 
 <SRS0=89Gv=LM=lists.infradead.org=linux-rockchip-bounces+linux-rockchip=archiver.kernel.org@kernel.org>
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
	aws-us-west-2-korg-lkml-1.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-13.7 required=3.0 tests=BAYES_00,
	DKIM_ADSP_CUSTOM_MED,DKIM_SIGNED,DKIM_VALID,FREEMAIL_FORGED_FROMDOMAIN,
	FREEMAIL_FROM,HEADER_FROM_DIFFERENT_DOMAINS,INCLUDES_CR_TRAILER,
	INCLUDES_PATCH,MAILING_LIST_MULTI,SPF_HELO_NONE,SPF_PASS,URIBL_BLOCKED,
	USER_AGENT_GIT autolearn=ham autolearn_force=no version=3.4.0
Received: from mail.kernel.org (mail.kernel.org [198.145.29.99])
	by smtp.lore.kernel.org (Postfix) with ESMTP id D3138C48BDF
	for <linux-rockchip@archiver.kernel.org>;
 Fri, 18 Jun 2021 18:13:21 +0000 (UTC)
Received: from bombadil.infradead.org (bombadil.infradead.org
 [198.137.202.133])
	(using TLSv1.2 with cipher ECDHE-RSA-AES256-GCM-SHA384 (256/256 bits))
	(No client certificate requested)
	by mail.kernel.org (Postfix) with ESMTPS id 94EF2613ED
	for <linux-rockchip@archiver.kernel.org>;
 Fri, 18 Jun 2021 18:13:21 +0000 (UTC)
DMARC-Filter: OpenDMARC Filter v1.3.2 mail.kernel.org 94EF2613ED
Authentication-Results: mail.kernel.org;
 dmarc=fail (p=none dis=none) header.from=gmail.com
Authentication-Results: mail.kernel.org;
 spf=none
 smtp.mailfrom=linux-rockchip-bounces+linux-rockchip=archiver.kernel.org@lists.infradead.org
DKIM-Signature: v=1; a=rsa-sha256; q=dns/txt; c=relaxed/relaxed;
	d=lists.infradead.org; s=bombadil.20210309; h=Sender:
	Content-Transfer-Encoding:Content-Type:List-Subscribe:List-Help:List-Post:
	List-Archive:List-Unsubscribe:List-Id:MIME-Version:References:In-Reply-To:
	Message-Id:Date:Subject:Cc:To:From:Reply-To:Content-ID:Content-Description:
	Resent-Date:Resent-From:Resent-Sender:Resent-To:Resent-Cc:Resent-Message-ID:
	List-Owner; bh=qX5QtWNAWObQJcjgEidxCIFUChiUlrwH2ABHgIHM66A=; b=zLCtCN9AfHob0a
	QZkunTnXFKgjccZSqgJLUn25sY56lnEOoXUln5KXS3hYFVhA2op+vjd7c/M1RjBhDQKXDjDKjKD2N
	c5ipsLNM2ZCSPR7uKuO8wCojxPpRlJO+fFAZmWyjr73BJ7XOCED56HzUAd4CwB+kNbiH6vSrPIwYv
	ddcqrSnf5I8aqeCjm3TZ63YMWY6emX/21wQsVN20pfldFb4cE6n4mXjuxnfNvuWXLzZ1u4l637yaF
	imwigx/PEPifncUdnhTv6aSZBS3KcS9Nz5sipolfTyGA32XekxL+iHIbzv/Awh8fgHaBJRDjbh4XT
	/ExViSW8CGm0K1kNUnWg==;
Received: from localhost ([::1] helo=bombadil.infradead.org)
	by bombadil.infradead.org with esmtp (Exim 4.94.2 #2 (Red Hat Linux))
	id 1luIzV-00F7xA-Qb; Fri, 18 Jun 2021 18:13:17 +0000
Received: from mail-wr1-x42a.google.com ([2a00:1450:4864:20::42a])
 by bombadil.infradead.org with esmtps (Exim 4.94.2 #2 (Red Hat Linux))
 id 1luIzN-00F7vF-En; Fri, 18 Jun 2021 18:13:10 +0000
Received: by mail-wr1-x42a.google.com with SMTP id n7so11709160wri.3;
 Fri, 18 Jun 2021 11:13:08 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=gmail.com; s=20161025;
 h=from:to:cc:subject:date:message-id:in-reply-to:references
 :mime-version:content-transfer-encoding;
 bh=htjXApkhNdnP2TPogXqT3DcJXBVLwZ/N/O4uwhluccE=;
 b=P94bMjEVAhghS3e8W5TNzb5y1EnmV4DRB8X6rO4hSox87ZbVbQe8DH8tfDUeeh9Si6
 SgP02Y/+cLTYVJoFUAXvj9rkIMMeLq4Cm8ajMrRwVLi9ZX5xFxOIUfyW+ALpFzpUXnlF
 Rg8uxa0VeNtO4M3DqbOeJkJ6OF2D+CSEdh04mbsjBpJP6d0xjHVueqHKkujC0LlHXhVV
 lKdLiS9l0WVdfS9aa/qKqDN1447W5KzbGAjOd+igh9YZElxtQ31BiHGdWbrSgaVTdaQ4
 bqzcUsOiciLxvNUue4Zxzc4EEYD8OPSQGEaTt/5gAs5gbmB6uhtBZsU1P24U7/S0hmRI
 mjQQ==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
 d=1e100.net; s=20161025;
 h=x-gm-message-state:from:to:cc:subject:date:message-id:in-reply-to
 :references:mime-version:content-transfer-encoding;
 bh=htjXApkhNdnP2TPogXqT3DcJXBVLwZ/N/O4uwhluccE=;
 b=L54VngYyOJc8eylgESZt94k4R9gR0hTLnUE/YjA3/GJjItAstKqqS48VhonB46Z/OW
 6QGoClMLWnuiOLGTvW5WK1PuNttVLiHSaiGGtDho8cm/5HptOM9wJF+nHXTYMjo26cvG
 tjC8HxTaCmHl9owt0JVmmYQdEiR+ZUTo0XkNXeC0Ag9dCn5yIg1pHnzPQ9VTbg+H/4yg
 TsVCwh2laCvsMG/cXyfcm9Mm0wYap7eKqzPmkMnS9XFD0JTvTbF3SdtYW+wxyWkFoEnV
 0XHUXkFGTLO0X8XLFe6U6cj9jScYjy1tCSYNiXRaqIC6pLnOtKT458gdASRYHUkXnpA/
 YfVg==
X-Gm-Message-State: AOAM533sbLdqq6l+J/xTYovHosQetb8Md/qoOaseDQgTVYUZ3nLg166d
 HIAjj6SAPeaXCUARtsiF7GQ33SkYcr0O
X-Google-Smtp-Source: 
 ABdhPJwhb8/oWvEHsp+eeJJ3vSb744SlgxAnekBNoPjdu6MG4ouNSzdrlSrWY2hLX/GtKYAMb+swTg==
X-Received: by 2002:a5d:6acc:: with SMTP id
 u12mr14032258wrw.414.1624039987716;
 Fri, 18 Jun 2021 11:13:07 -0700 (PDT)
Received: from localhost.localdomain (ip5b434b8b.dynamic.kabel-deutschland.de.
 [91.67.75.139])
 by smtp.googlemail.com with ESMTPSA id l10sm9306782wrv.82.2021.06.18.11.13.07
 (version=TLS1_3 cipher=TLS_AES_256_GCM_SHA384 bits=256/256);
 Fri, 18 Jun 2021 11:13:07 -0700 (PDT)
From: Alex Bee <knaerzche@gmail.com>
To: Heiko Stuebner <heiko@sntech.de>,
	Rob Herring <robh+dt@kernel.org>
Cc: Johan Jonker <jbx6244@gmail.com>, linux-rockchip@lists.infradead.org,
 devicetree@vger.kernel.org, linux-arm-kernel@lists.infradead.org,
 linux-kernel@vger.kernel.org, Alex Bee <knaerzche@gmail.com>
Subject: [PATCH v2 1/5] dt-bindings: Add doc for ROCK Pi 4 A+ and B+
Date: Fri, 18 Jun 2021 20:12:52 +0200
Message-Id: <20210618181256.27992-2-knaerzche@gmail.com>
X-Mailer: git-send-email 2.27.0
In-Reply-To: <20210618181256.27992-1-knaerzche@gmail.com>
References: <20210618181256.27992-1-knaerzche@gmail.com>
MIME-Version: 1.0
X-CRM114-Version: 20100106-BlameMichelson ( TRE 0.8.0 (BSD) ) MR-646709E3 
X-CRM114-CacheID: sfid-20210618_111309_548439_1A9382AC 
X-CRM114-Status: GOOD (  11.07  )
X-BeenThere: linux-rockchip@lists.infradead.org
X-Mailman-Version: 2.1.34
Precedence: list
List-Id: Upstream kernel work for Rockchip platforms
 <linux-rockchip.lists.infradead.org>
List-Unsubscribe: <http://lists.infradead.org/mailman/options/linux-rockchip>,
 <mailto:linux-rockchip-request@lists.infradead.org?subject=unsubscribe>
List-Archive: <http://lists.infradead.org/pipermail/linux-rockchip/>
List-Post: <mailto:linux-rockchip@lists.infradead.org>
List-Help: <mailto:linux-rockchip-request@lists.infradead.org?subject=help>
List-Subscribe: <http://lists.infradead.org/mailman/listinfo/linux-rockchip>,
 <mailto:linux-rockchip-request@lists.infradead.org?subject=subscribe>
Sender: "Linux-rockchip" <linux-rockchip-bounces@lists.infradead.org>
Errors-To: 
 linux-rockchip-bounces+linux-rockchip=archiver.kernel.org@lists.infradead.org

ROCK Pi 4 got 2 more variants called A+ and B+.
Add the dt-bindings documentation for it.

Signed-off-by: Alex Bee <knaerzche@gmail.com>
Acked-by: Rob Herring <robh@kernel.org>
---
 Documentation/devicetree/bindings/arm/rockchip.yaml | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/Documentation/devicetree/bindings/arm/rockchip.yaml b/Documentation/devicetree/bindings/arm/rockchip.yaml
index 4a6f772c1043..e5049550bdbf 100644
--- a/Documentation/devicetree/bindings/arm/rockchip.yaml
+++ b/Documentation/devicetree/bindings/arm/rockchip.yaml
@@ -460,11 +460,13 @@ properties:
           - const: radxa,rock
           - const: rockchip,rk3188
 
-      - description: Radxa ROCK Pi 4A/B/C
+      - description: Radxa ROCK Pi 4A/A+/B/B+/C
         items:
           - enum:
               - radxa,rockpi4a
+              - radxa,rockpi4a-plus
               - radxa,rockpi4b
+              - radxa,rockpi4b-plus
               - radxa,rockpi4c
           - const: radxa,rockpi4
           - const: rockchip,rk3399

From patchwork Fri Jun 18 18:12:53 2021
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Alex Bee <knaerzche@gmail.com>
X-Patchwork-Id: 12331873
Return-Path: 
 <SRS0=89Gv=LM=lists.infradead.org=linux-rockchip-bounces+linux-rockchip=archiver.kernel.org@kernel.org>
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
	aws-us-west-2-korg-lkml-1.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-13.7 required=3.0 tests=BAYES_00,
	DKIM_ADSP_CUSTOM_MED,DKIM_SIGNED,DKIM_VALID,FREEMAIL_FORGED_FROMDOMAIN,
	FREEMAIL_FROM,HEADER_FROM_DIFFERENT_DOMAINS,INCLUDES_CR_TRAILER,
	INCLUDES_PATCH,MAILING_LIST_MULTI,SPF_HELO_NONE,SPF_PASS,URIBL_BLOCKED,
	USER_AGENT_GIT autolearn=ham autolearn_force=no version=3.4.0
Received: from mail.kernel.org (mail.kernel.org [198.145.29.99])
	by smtp.lore.kernel.org (Postfix) with ESMTP id C6A85C49361
	for <linux-rockchip@archiver.kernel.org>;
 Fri, 18 Jun 2021 18:13:33 +0000 (UTC)
Received: from bombadil.infradead.org (bombadil.infradead.org
 [198.137.202.133])
	(using TLSv1.2 with cipher ECDHE-RSA-AES256-GCM-SHA384 (256/256 bits))
	(No client certificate requested)
	by mail.kernel.org (Postfix) with ESMTPS id 664FD613ED
	for <linux-rockchip@archiver.kernel.org>;
 Fri, 18 Jun 2021 18:13:33 +0000 (UTC)
DMARC-Filter: OpenDMARC Filter v1.3.2 mail.kernel.org 664FD613ED
Authentication-Results: mail.kernel.org;
 dmarc=fail (p=none dis=none) header.from=gmail.com
Authentication-Results: mail.kernel.org;
 spf=none
 smtp.mailfrom=linux-rockchip-bounces+linux-rockchip=archiver.kernel.org@lists.infradead.org
DKIM-Signature: v=1; a=rsa-sha256; q=dns/txt; c=relaxed/relaxed;
	d=lists.infradead.org; s=bombadil.20210309; h=Sender:
	Content-Transfer-Encoding:Content-Type:List-Subscribe:List-Help:List-Post:
	List-Archive:List-Unsubscribe:List-Id:MIME-Version:References:In-Reply-To:
	Message-Id:Date:Subject:Cc:To:From:Reply-To:Content-ID:Content-Description:
	Resent-Date:Resent-From:Resent-Sender:Resent-To:Resent-Cc:Resent-Message-ID:
	List-Owner; bh=tRMrO77w8GOXSRinvWM7sGF3/RPQiIoRtlr3Ki1WYUY=; b=ESxliFB3YN9xpF
	vFWo3DCea4/xEepgDCySq+lED32BkIIdX8D5bG6x2Qrpk8xHxgS7UAj+vOqTPzuN1is8j7l3F6Mm0
	SToQ7abXt34ttjy+vlijw3DGaFE+x0yCM8TCbvsM40FMJFxTiYSTi/tQs6KNsuTU343TMqrEwc7Fj
	KQOWNo/VRrRNY9Y4tKBVZTNEdnr6CNQrk7lWY2vrVJcoTKmZkFSREYj3BQWgk+ugdK6kQAaDGDpZA
	f+tNKBZyhmiDB5mzY1w9p5I9bxaN6Ax2Oq9WcWxCcqUeI9blURD4yZQBCCZ3E72n/NQB+GyoIcMLl
	pO5iMOqsGlqgqNTViqfA==;
Received: from localhost ([::1] helo=bombadil.infradead.org)
	by bombadil.infradead.org with esmtp (Exim 4.94.2 #2 (Red Hat Linux))
	id 1luIzi-00F82K-0y; Fri, 18 Jun 2021 18:13:30 +0000
Received: from mail-wm1-x335.google.com ([2a00:1450:4864:20::335])
 by bombadil.infradead.org with esmtps (Exim 4.94.2 #2 (Red Hat Linux))
 id 1luIzV-00F7wf-5z; Fri, 18 Jun 2021 18:13:18 +0000
Received: by mail-wm1-x335.google.com with SMTP id
 o39-20020a05600c5127b02901d23584fd9bso6457869wms.0;
 Fri, 18 Jun 2021 11:13:16 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=gmail.com; s=20161025;
 h=from:to:cc:subject:date:message-id:in-reply-to:references
 :mime-version:content-transfer-encoding;
 bh=rv4VKDRBcNrskkvEnd89Jswu3eRnd7ncejFN962fJIc=;
 b=MliDCA4h82Qv6GO6z0JtwJO5+8tXwdfODNzdyQKaMkLkFWiFf8n4HuqoMCeTeEW+YE
 B0UMnx53hRwiXJbZGkfdTK1zdrpS4b1m+rCz1hTKBPi2UW7itn2L8elTxo3sg5Du8k6C
 BQ8K/X6GuIwXrpaXqljV5OBiURiZpdgYo8f2ExCjzGAdKTvcsI5EWiNkecEmB7wBug71
 kmiMWKt4JzgM1FIODJd2Or8o0mhttgwr/vHyBc7s7eUnbwZjvYc+1s9p73wlzatQKK59
 5jQ82e/GTIVpSI0YolaPFIKxpIcofiQMUo3EqRrWCBpC3bdUv+GLti9jhHb0jNgZR+my
 O0lw==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
 d=1e100.net; s=20161025;
 h=x-gm-message-state:from:to:cc:subject:date:message-id:in-reply-to
 :references:mime-version:content-transfer-encoding;
 bh=rv4VKDRBcNrskkvEnd89Jswu3eRnd7ncejFN962fJIc=;
 b=oJT2xgDxX6Ec/PRW+ap39nKXpAOfsfhfVv/v8qH+HP0h5AbJQBeMlcxBcKXvjgiolf
 9pC3TAOBDcL51MfEqHggbVLvTHoOhMZZCUo23WxOg+AaF8vNKJJxjtHFoJDcPoEtPcMV
 nR7w/5KWBqtXe5gfCOXBvigvjZwtfrKbZUCjf/rlaNHzGp+q7BlU51n6sUss9W1kyV7z
 eb9xBp/Qx9eHu1Z8jerIft3lDHfnr9zKFpaujKU5I00imf6iO2VqNeLNoF6qTbMcd8p9
 Tnu+joDB4CIqP2uSH+iN1P8Dk3Q/2aXST9Q7XYzGMCi6oSBDMGGLPEmGwyMmWkbZ32Vd
 PKHg==
X-Gm-Message-State: AOAM530YEIHYvTnoW+3TcYcZX1FKw0zQoxOuu/9L70/crZD2NH3CWAj7
 m8uFVMC173iR7+XJRwPMuw==
X-Google-Smtp-Source: 
 ABdhPJw61tE1SgD3KZkdhjmX83akHP4VSX6zBLhC8lhARwMxkLIsxlU5wpu1/jpGew1IKacMRmYctA==
X-Received: by 2002:a05:600c:21d0:: with SMTP id
 x16mr2771318wmj.78.1624039995461;
 Fri, 18 Jun 2021 11:13:15 -0700 (PDT)
Received: from localhost.localdomain (ip5b434b8b.dynamic.kabel-deutschland.de.
 [91.67.75.139])
 by smtp.googlemail.com with ESMTPSA id l10sm9306782wrv.82.2021.06.18.11.13.14
 (version=TLS1_3 cipher=TLS_AES_256_GCM_SHA384 bits=256/256);
 Fri, 18 Jun 2021 11:13:15 -0700 (PDT)
From: Alex Bee <knaerzche@gmail.com>
To: Heiko Stuebner <heiko@sntech.de>,
	Rob Herring <robh+dt@kernel.org>
Cc: Johan Jonker <jbx6244@gmail.com>, linux-rockchip@lists.infradead.org,
 devicetree@vger.kernel.org, linux-arm-kernel@lists.infradead.org,
 linux-kernel@vger.kernel.org, Alex Bee <knaerzche@gmail.com>
Subject: [PATCH v2 2/5] arm64: dts: rockchip: Add RK3399 ROCK Pi 4A+ board
Date: Fri, 18 Jun 2021 20:12:53 +0200
Message-Id: <20210618181256.27992-3-knaerzche@gmail.com>
X-Mailer: git-send-email 2.27.0
In-Reply-To: <20210618181256.27992-1-knaerzche@gmail.com>
References: <20210618181256.27992-1-knaerzche@gmail.com>
MIME-Version: 1.0
X-CRM114-Version: 20100106-BlameMichelson ( TRE 0.8.0 (BSD) ) MR-646709E3 
X-CRM114-CacheID: sfid-20210618_111317_271535_79E7AECF 
X-CRM114-Status: GOOD (  15.66  )
X-BeenThere: linux-rockchip@lists.infradead.org
X-Mailman-Version: 2.1.34
Precedence: list
List-Id: Upstream kernel work for Rockchip platforms
 <linux-rockchip.lists.infradead.org>
List-Unsubscribe: <http://lists.infradead.org/mailman/options/linux-rockchip>,
 <mailto:linux-rockchip-request@lists.infradead.org?subject=unsubscribe>
List-Archive: <http://lists.infradead.org/pipermail/linux-rockchip/>
List-Post: <mailto:linux-rockchip@lists.infradead.org>
List-Help: <mailto:linux-rockchip-request@lists.infradead.org?subject=help>
List-Subscribe: <http://lists.infradead.org/mailman/listinfo/linux-rockchip>,
 <mailto:linux-rockchip-request@lists.infradead.org?subject=subscribe>
Sender: "Linux-rockchip" <linux-rockchip-bounces@lists.infradead.org>
Errors-To: 
 linux-rockchip-bounces+linux-rockchip=archiver.kernel.org@lists.infradead.org

ROCK Pi 4A+ board is the successor of ROCK Pi 4A board.

Differences to the original version are
- has RK3399 OP1 SoC revision
- has eMMC (16 or 32 GB) soldered on board (no changes required,
  since it is enabled in rk3399-rock-pi-4.dtsi)
- dev boards have SPI flash soldered, but as per manufacturer response,
  this won't be the case for mass production boards

Signed-off-by: Alex Bee <knaerzche@gmail.com>
---
 arch/arm64/boot/dts/rockchip/Makefile              |  1 +
 .../boot/dts/rockchip/rk3399-rock-pi-4a-plus.dts   | 14 ++++++++++++++
 2 files changed, 15 insertions(+)
 create mode 100644 arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4a-plus.dts

diff --git a/arch/arm64/boot/dts/rockchip/Makefile b/arch/arm64/boot/dts/rockchip/Makefile
index c3e00c0e2db7..dbd7d37950f1 100644
--- a/arch/arm64/boot/dts/rockchip/Makefile
+++ b/arch/arm64/boot/dts/rockchip/Makefile
@@ -43,6 +43,7 @@ dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3399-puma-haikou.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3399-roc-pc.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3399-roc-pc-mezzanine.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3399-rock-pi-4a.dtb
+dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3399-rock-pi-4a-plus.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3399-rock-pi-4b.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3399-rock-pi-4c.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3399-rock960.dtb
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4a-plus.dts b/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4a-plus.dts
new file mode 100644
index 000000000000..281a04b2f5e9
--- /dev/null
+++ b/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4a-plus.dts
@@ -0,0 +1,14 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Copyright (c) 2019 Akash Gajjar <Akash_Gajjar@mentor.com>
+ * Copyright (c) 2019 Pragnesh Patel <Pragnesh_Patel@mentor.com>
+ */
+
+/dts-v1/;
+#include "rk3399-rock-pi-4.dtsi"
+#include "rk3399-op1-opp.dtsi"
+
+/ {
+	model = "Radxa ROCK Pi 4A+";
+	compatible = "radxa,rockpi4a-plus", "radxa,rockpi4", "rockchip,rk3399";
+};

From patchwork Fri Jun 18 18:12:54 2021
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Alex Bee <knaerzche@gmail.com>
X-Patchwork-Id: 12331875
Return-Path: 
 <SRS0=89Gv=LM=lists.infradead.org=linux-rockchip-bounces+linux-rockchip=archiver.kernel.org@kernel.org>
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
	aws-us-west-2-korg-lkml-1.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-13.7 required=3.0 tests=BAYES_00,
	DKIM_ADSP_CUSTOM_MED,DKIM_SIGNED,DKIM_VALID,FREEMAIL_FORGED_FROMDOMAIN,
	FREEMAIL_FROM,HEADER_FROM_DIFFERENT_DOMAINS,INCLUDES_CR_TRAILER,
	INCLUDES_PATCH,MAILING_LIST_MULTI,SPF_HELO_NONE,SPF_PASS,URIBL_BLOCKED,
	USER_AGENT_GIT autolearn=unavailable autolearn_force=no version=3.4.0
Received: from mail.kernel.org (mail.kernel.org [198.145.29.99])
	by smtp.lore.kernel.org (Postfix) with ESMTP id 4F338C48BDF
	for <linux-rockchip@archiver.kernel.org>;
 Fri, 18 Jun 2021 18:13:47 +0000 (UTC)
Received: from bombadil.infradead.org (bombadil.infradead.org
 [198.137.202.133])
	(using TLSv1.2 with cipher ECDHE-RSA-AES256-GCM-SHA384 (256/256 bits))
	(No client certificate requested)
	by mail.kernel.org (Postfix) with ESMTPS id 12A11613ED
	for <linux-rockchip@archiver.kernel.org>;
 Fri, 18 Jun 2021 18:13:47 +0000 (UTC)
DMARC-Filter: OpenDMARC Filter v1.3.2 mail.kernel.org 12A11613ED
Authentication-Results: mail.kernel.org;
 dmarc=fail (p=none dis=none) header.from=gmail.com
Authentication-Results: mail.kernel.org;
 spf=none
 smtp.mailfrom=linux-rockchip-bounces+linux-rockchip=archiver.kernel.org@lists.infradead.org
DKIM-Signature: v=1; a=rsa-sha256; q=dns/txt; c=relaxed/relaxed;
	d=lists.infradead.org; s=bombadil.20210309; h=Sender:
	Content-Transfer-Encoding:Content-Type:List-Subscribe:List-Help:List-Post:
	List-Archive:List-Unsubscribe:List-Id:MIME-Version:References:In-Reply-To:
	Message-Id:Date:Subject:Cc:To:From:Reply-To:Content-ID:Content-Description:
	Resent-Date:Resent-From:Resent-Sender:Resent-To:Resent-Cc:Resent-Message-ID:
	List-Owner; bh=hQiZj712xoOPRXcnQGQqK1k+VdwsP+KcYHvWWcFiTTE=; b=CS9Jr1Soxzh+ur
	h+28Z7TiBJAZh46kSkr3Z9e6TUDutdmNmVRHhijsju/JAkir+t/9152UkNVteZk8wNkwWhsAnXnKw
	KhCp3eAUYPpxt/h9WAXE43aIh7ePV10TLlF/OofMma2LgqsHm1N5LmtPBaTec5BEgs86943MFZoBw
	fy9LM9ZAYFwCNm4kgSMT2gbO0hExvu0MN1sUfKRBsFCKbrTnm477HX69STv8hVItKXCGWwEvl9TQ5
	jPZc6mQ4kEolXk5hIKwqzMfpmWArS3K3LaHFVzIauqT5e65KCvBSbR40f2A+Ih8Q3zcY9cN2Jxhf2
	Ktk/AWOehvuwvQhEFCGA==;
Received: from localhost ([::1] helo=bombadil.infradead.org)
	by bombadil.infradead.org with esmtp (Exim 4.94.2 #2 (Red Hat Linux))
	id 1luIzv-00F87v-JF; Fri, 18 Jun 2021 18:13:43 +0000
Received: from mail-wr1-x433.google.com ([2a00:1450:4864:20::433])
 by bombadil.infradead.org with esmtps (Exim 4.94.2 #2 (Red Hat Linux))
 id 1luIzb-00F7yx-E4; Fri, 18 Jun 2021 18:13:24 +0000
Received: by mail-wr1-x433.google.com with SMTP id e22so8114153wrc.1;
 Fri, 18 Jun 2021 11:13:22 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=gmail.com; s=20161025;
 h=from:to:cc:subject:date:message-id:in-reply-to:references
 :mime-version:content-transfer-encoding;
 bh=kn98Ll5NtpVpipbgUQvn4hSMRQNojtQUAei8tOm3g58=;
 b=NIX0oLDftDi6biHsi+LL0gntF8jwaNG+amkxSzoGyPBQI7xlAmcR3NfrZ54USdFQP4
 q1QegsItPIubfkqLRpRe+0HpO9mszszJ9Br7N+0RkFmDUCcgKnGiRMrq/QfCHZlMIAzj
 AAWcjADKNmqy6kcwwvqI2Km/yTMcdaRIkrxiBw+VwKZe9Sf7Q7D87H7ddAr195ElfWQH
 k3OgTjecka+toNHRxZXYD5NQ+re7SluV3OXE6xGbKKRkktxz8qRoaYIXLVXMxAYYcpcd
 +LrCFT8+nLsvOdnXT3gSrACevK8+AbHb13URNoQaCmz8sf0VHBeR6gnUPWS0YiY36qmH
 8qNg==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
 d=1e100.net; s=20161025;
 h=x-gm-message-state:from:to:cc:subject:date:message-id:in-reply-to
 :references:mime-version:content-transfer-encoding;
 bh=kn98Ll5NtpVpipbgUQvn4hSMRQNojtQUAei8tOm3g58=;
 b=ulInyqpyIFm3GdYTvkxVJg2yWMESaqV2zpTpbdD3lloX2o2Pe5pRD3izOYJhId3L+p
 0bEYatooqSLn7q8EDSBXh90L47AUYjy+1XcZt/hQ+0oXFVXX0PhEDFAGHvEteWfEIjqF
 +MMdbx2UYxLThzfs9KHEsVio8IqfxfuWjZz00EjX1evbiNuu6JOPs7VkeGSNJca01zKf
 OBWUGEL51oNt5Mna0zH1iCSDP9gyb/eRYt0n2o+BstmkFgFNk6MJSTCD1bMo9IKI3n1M
 J0I58l2kZc5eM0/yM10h+e1qwQlO5hT2fXE2OSUIRYpv0ZuDXjHuu28nhViTLjuyoGbC
 SNtA==
X-Gm-Message-State: AOAM5325w1a0KWtKKrhhU8/3EMTjkQHbeD8tHovATmQwvvW3uBDzfqYW
 AcKjGaFfNoL+6grBkY2qQw==
X-Google-Smtp-Source: 
 ABdhPJz8Y2F3LWyjxmZDk8cpgSz9/LhbtqqL7qNCHFkVV4MLCnbXc5uPbiQj0rJ2h/sY6FuHztbgYA==
X-Received: by 2002:adf:e502:: with SMTP id j2mr7001313wrm.275.1624040001599;
 Fri, 18 Jun 2021 11:13:21 -0700 (PDT)
Received: from localhost.localdomain (ip5b434b8b.dynamic.kabel-deutschland.de.
 [91.67.75.139])
 by smtp.googlemail.com with ESMTPSA id l10sm9306782wrv.82.2021.06.18.11.13.21
 (version=TLS1_3 cipher=TLS_AES_256_GCM_SHA384 bits=256/256);
 Fri, 18 Jun 2021 11:13:21 -0700 (PDT)
From: Alex Bee <knaerzche@gmail.com>
To: Heiko Stuebner <heiko@sntech.de>,
	Rob Herring <robh+dt@kernel.org>
Cc: Johan Jonker <jbx6244@gmail.com>, linux-rockchip@lists.infradead.org,
 devicetree@vger.kernel.org, linux-arm-kernel@lists.infradead.org,
 linux-kernel@vger.kernel.org, Alex Bee <knaerzche@gmail.com>
Subject: [PATCH v2 3/5] arm64: dts: rockchip: Add RK3399 ROCK Pi 4B+ board
Date: Fri, 18 Jun 2021 20:12:54 +0200
Message-Id: <20210618181256.27992-4-knaerzche@gmail.com>
X-Mailer: git-send-email 2.27.0
In-Reply-To: <20210618181256.27992-1-knaerzche@gmail.com>
References: <20210618181256.27992-1-knaerzche@gmail.com>
MIME-Version: 1.0
X-CRM114-Version: 20100106-BlameMichelson ( TRE 0.8.0 (BSD) ) MR-646709E3 
X-CRM114-CacheID: sfid-20210618_111323_517365_E75A7CF8 
X-CRM114-Status: GOOD (  15.84  )
X-BeenThere: linux-rockchip@lists.infradead.org
X-Mailman-Version: 2.1.34
Precedence: list
List-Id: Upstream kernel work for Rockchip platforms
 <linux-rockchip.lists.infradead.org>
List-Unsubscribe: <http://lists.infradead.org/mailman/options/linux-rockchip>,
 <mailto:linux-rockchip-request@lists.infradead.org?subject=unsubscribe>
List-Archive: <http://lists.infradead.org/pipermail/linux-rockchip/>
List-Post: <mailto:linux-rockchip@lists.infradead.org>
List-Help: <mailto:linux-rockchip-request@lists.infradead.org?subject=help>
List-Subscribe: <http://lists.infradead.org/mailman/listinfo/linux-rockchip>,
 <mailto:linux-rockchip-request@lists.infradead.org?subject=subscribe>
Sender: "Linux-rockchip" <linux-rockchip-bounces@lists.infradead.org>
Errors-To: 
 linux-rockchip-bounces+linux-rockchip=archiver.kernel.org@lists.infradead.org

ROCK Pi 4B+ board is the successor of ROCK Pi 4B board.

Differences to the original version are
- has RK3399 OP1 SoC revision
- has eMMC (16 or 32 GB) soldered on board (no changes required,
  since it is enabled in rk3399-rock-pi-4.dtsi)
- dev boards have SPI flash soldered, but as per manufacturer response,
  this won't be the case for mass production boards

Signed-off-by: Alex Bee <knaerzche@gmail.com>
---
 arch/arm64/boot/dts/rockchip/Makefile         |  1 +
 .../dts/rockchip/rk3399-rock-pi-4b-plus.dts   | 47 +++++++++++++++++++
 2 files changed, 48 insertions(+)
 create mode 100644 arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4b-plus.dts

diff --git a/arch/arm64/boot/dts/rockchip/Makefile b/arch/arm64/boot/dts/rockchip/Makefile
index dbd7d37950f1..e867e5889d44 100644
--- a/arch/arm64/boot/dts/rockchip/Makefile
+++ b/arch/arm64/boot/dts/rockchip/Makefile
@@ -45,6 +45,7 @@ dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3399-roc-pc-mezzanine.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3399-rock-pi-4a.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3399-rock-pi-4a-plus.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3399-rock-pi-4b.dtb
+dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3399-rock-pi-4b-plus.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3399-rock-pi-4c.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3399-rock960.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3399-rockpro64-v2.dtb
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4b-plus.dts b/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4b-plus.dts
new file mode 100644
index 000000000000..dfad13d2ab24
--- /dev/null
+++ b/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4b-plus.dts
@@ -0,0 +1,47 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Copyright (c) 2019 Akash Gajjar <Akash_Gajjar@mentor.com>
+ * Copyright (c) 2019 Pragnesh Patel <Pragnesh_Patel@mentor.com>
+ */
+
+/dts-v1/;
+#include "rk3399-rock-pi-4.dtsi"
+#include "rk3399-op1-opp.dtsi"
+
+/ {
+	model = "Radxa ROCK Pi 4B+";
+	compatible = "radxa,rockpi4b-plus", "radxa,rockpi4", "rockchip,rk3399";
+
+	aliases {
+		mmc2 = &sdio0;
+	};
+};
+
+&sdio0 {
+	status = "okay";
+
+	brcmf: wifi@1 {
+		compatible = "brcm,bcm4329-fmac";
+		reg = <1>;
+		interrupt-parent = <&gpio0>;
+		interrupts = <RK_PA3 GPIO_ACTIVE_HIGH>;
+		interrupt-names = "host-wake";
+		pinctrl-names = "default";
+		pinctrl-0 = <&wifi_host_wake_l>;
+	};
+};
+
+&uart0 {
+	status = "okay";
+
+	bluetooth {
+		compatible = "brcm,bcm43438-bt";
+		clocks = <&rk808 1>;
+		clock-names = "ext_clock";
+		device-wakeup-gpios = <&gpio2 RK_PD3 GPIO_ACTIVE_HIGH>;
+		host-wakeup-gpios = <&gpio0 RK_PA4 GPIO_ACTIVE_HIGH>;
+		shutdown-gpios = <&gpio0 RK_PB1 GPIO_ACTIVE_HIGH>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&bt_host_wake_l &bt_wake_l &bt_enable_h>;
+	};
+};

From patchwork Fri Jun 18 18:12:56 2021
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Alex Bee <knaerzche@gmail.com>
X-Patchwork-Id: 12331879
Return-Path: 
 <SRS0=89Gv=LM=lists.infradead.org=linux-rockchip-bounces+linux-rockchip=archiver.kernel.org@kernel.org>
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
	aws-us-west-2-korg-lkml-1.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-13.7 required=3.0 tests=BAYES_00,
	DKIM_ADSP_CUSTOM_MED,DKIM_SIGNED,DKIM_VALID,FREEMAIL_FORGED_FROMDOMAIN,
	FREEMAIL_FROM,HEADER_FROM_DIFFERENT_DOMAINS,INCLUDES_CR_TRAILER,
	INCLUDES_PATCH,MAILING_LIST_MULTI,SPF_HELO_NONE,SPF_PASS,URIBL_BLOCKED,
	USER_AGENT_GIT autolearn=ham autolearn_force=no version=3.4.0
Received: from mail.kernel.org (mail.kernel.org [198.145.29.99])
	by smtp.lore.kernel.org (Postfix) with ESMTP id 0CD52C48BE8
	for <linux-rockchip@archiver.kernel.org>;
 Fri, 18 Jun 2021 18:14:22 +0000 (UTC)
Received: from bombadil.infradead.org (bombadil.infradead.org
 [198.137.202.133])
	(using TLSv1.2 with cipher ECDHE-RSA-AES256-GCM-SHA384 (256/256 bits))
	(No client certificate requested)
	by mail.kernel.org (Postfix) with ESMTPS id CA8A86127C
	for <linux-rockchip@archiver.kernel.org>;
 Fri, 18 Jun 2021 18:14:21 +0000 (UTC)
DMARC-Filter: OpenDMARC Filter v1.3.2 mail.kernel.org CA8A86127C
Authentication-Results: mail.kernel.org;
 dmarc=fail (p=none dis=none) header.from=gmail.com
Authentication-Results: mail.kernel.org;
 spf=none
 smtp.mailfrom=linux-rockchip-bounces+linux-rockchip=archiver.kernel.org@lists.infradead.org
DKIM-Signature: v=1; a=rsa-sha256; q=dns/txt; c=relaxed/relaxed;
	d=lists.infradead.org; s=bombadil.20210309; h=Sender:
	Content-Transfer-Encoding:Content-Type:List-Subscribe:List-Help:List-Post:
	List-Archive:List-Unsubscribe:List-Id:MIME-Version:References:In-Reply-To:
	Message-Id:Date:Subject:Cc:To:From:Reply-To:Content-ID:Content-Description:
	Resent-Date:Resent-From:Resent-Sender:Resent-To:Resent-Cc:Resent-Message-ID:
	List-Owner; bh=iPsj+ZvhB8xGUVuSOn+69KMiSxLCXXys0xE1LZqcUcI=; b=tE0tcULZ89PfEg
	Bg6jHlZt3aVsPQJRxOeO5mgaH3RgtXjQkZ6GczvxGiDjpYDo2268eOLeRDAjFOhmql9eACG6LpKjs
	ymMs5AnzyplpqjMiXTFsFpMeo6Wji56uxhpe7Ie9F1MIJjQz0QO43xVuxBOItFbX2/v6GsFoSOPFc
	OFMdkrvhwI1SVRj99wnSh+pnkqVCuU8pWnImZ5ozeQpNQO6s6hjn+XqbDN4Plyyl5IQ9Xrm6E2YIS
	c8X75ZOJDo7HgYUPBJ+f6oR+iYVWqp3xtcieD6xSU9ag32ud8lw4HSzknarlmEjU5TxBOQlPxx3OO
	aThNm36AOYalX0wPbo6w==;
Received: from localhost ([::1] helo=bombadil.infradead.org)
	by bombadil.infradead.org with esmtp (Exim 4.94.2 #2 (Red Hat Linux))
	id 1luJ0U-00F8PP-BG; Fri, 18 Jun 2021 18:14:18 +0000
Received: from mail-wr1-x431.google.com ([2a00:1450:4864:20::431])
 by bombadil.infradead.org with esmtps (Exim 4.94.2 #2 (Red Hat Linux))
 id 1luIzh-00F81M-K4; Fri, 18 Jun 2021 18:13:30 +0000
Received: by mail-wr1-x431.google.com with SMTP id m18so11698968wrv.2;
 Fri, 18 Jun 2021 11:13:28 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=gmail.com; s=20161025;
 h=from:to:cc:subject:date:message-id:in-reply-to:references
 :mime-version:content-transfer-encoding;
 bh=FRadAAJthazKXCs3EEyrRLqIIy1SFHZ81iHPWgmnJRI=;
 b=ny3LB6aXzm8w8C3+A9jf74tf+BJ986neHs79X76TCRusy5yLSp0LO3EWdBMg0srLNh
 KIeluN/0svlz2LiKKVdBMwWQ2tyMAriywQiW9cqmW/LaGjqjtfbb5d9ECq2MWH2UpXmA
 mE6DQTzTQIrkOc82FAufn5lZ0UHmzBzzD1yoHZVQcKa45Wx/8lkFLyAy0o/QBW0MQVj0
 myGns/y4E4h/zzLtvcglLgvzg/Hz2drPsNPJpYYhgYi86j3gEAvx12/a7cnDoZUI+2bT
 n013PLal7MkOMd3BP6gE7OervN1873OvazXnpI1urafITd3UmANw62uMjEBfKxdlGmwt
 jWSQ==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
 d=1e100.net; s=20161025;
 h=x-gm-message-state:from:to:cc:subject:date:message-id:in-reply-to
 :references:mime-version:content-transfer-encoding;
 bh=FRadAAJthazKXCs3EEyrRLqIIy1SFHZ81iHPWgmnJRI=;
 b=VCVZ3s6UQifCrB5asAc3biEQ1qGP3KxDgIshi0b64K/D3CTqBdVpWw2CIGMtNRGzLa
 182CWheEQB/sD6EMrYczkdudjJAaMxQ6EWL5QsGnBD8lnv5SSEHQPloNmY8u8gl8Gggr
 3uKwPI8P0k1/QoOMZnL2MXvz/tvSYAgWuyix50nrrqmZcLPjdPXqYVPZ7OSLICWZQcmG
 fjooMapMGeM73E2FtGhbKHI/bx8eaXlwv+uI1Nr0sVioA21y9kRbHmHkI4aooeP1v26c
 Fw8yrpBmKNmUMGkrM9TLyJA6a81hmau4G718fc56sRK1l0H7RKOy0bHnmx2X1P2BNIks
 uWgg==
X-Gm-Message-State: AOAM5325yy5QUXilLg46r5GIuo9nX5W4iCq/kKnov19o5LYzlAk1CwNR
 nA7fRNjFYrfgZBikOvaEtDozHHmOJcyg
X-Google-Smtp-Source: 
 ABdhPJxwYQRQMqECTGWplAPGQubZuPc+DAoMbIBgAWiqVlg7kAw1Gi3FbNatZS9UZrU2FM6mBdz28w==
X-Received: by 2002:adf:d4cc:: with SMTP id
 w12mr13871170wrk.216.1624040007816;
 Fri, 18 Jun 2021 11:13:27 -0700 (PDT)
Received: from localhost.localdomain (ip5b434b8b.dynamic.kabel-deutschland.de.
 [91.67.75.139])
 by smtp.googlemail.com with ESMTPSA id l10sm9306782wrv.82.2021.06.18.11.13.27
 (version=TLS1_3 cipher=TLS_AES_256_GCM_SHA384 bits=256/256);
 Fri, 18 Jun 2021 11:13:27 -0700 (PDT)
From: Alex Bee <knaerzche@gmail.com>
To: Heiko Stuebner <heiko@sntech.de>,
	Rob Herring <robh+dt@kernel.org>
Cc: Johan Jonker <jbx6244@gmail.com>, linux-rockchip@lists.infradead.org,
 devicetree@vger.kernel.org, linux-arm-kernel@lists.infradead.org,
 linux-kernel@vger.kernel.org, Alex Bee <knaerzche@gmail.com>
Subject: [PATCH v2 5/5] arm64: dts: rockchip: add SPDIF node for ROCK Pi 4
Date: Fri, 18 Jun 2021 20:12:56 +0200
Message-Id: <20210618181256.27992-6-knaerzche@gmail.com>
X-Mailer: git-send-email 2.27.0
In-Reply-To: <20210618181256.27992-1-knaerzche@gmail.com>
References: <20210618181256.27992-1-knaerzche@gmail.com>
MIME-Version: 1.0
X-CRM114-Version: 20100106-BlameMichelson ( TRE 0.8.0 (BSD) ) MR-646709E3 
X-CRM114-CacheID: sfid-20210618_111329_765501_D888146E 
X-CRM114-Status: GOOD (  12.96  )
X-BeenThere: linux-rockchip@lists.infradead.org
X-Mailman-Version: 2.1.34
Precedence: list
List-Id: Upstream kernel work for Rockchip platforms
 <linux-rockchip.lists.infradead.org>
List-Unsubscribe: <http://lists.infradead.org/mailman/options/linux-rockchip>,
 <mailto:linux-rockchip-request@lists.infradead.org?subject=unsubscribe>
List-Archive: <http://lists.infradead.org/pipermail/linux-rockchip/>
List-Post: <mailto:linux-rockchip@lists.infradead.org>
List-Help: <mailto:linux-rockchip-request@lists.infradead.org?subject=help>
List-Subscribe: <http://lists.infradead.org/mailman/listinfo/linux-rockchip>,
 <mailto:linux-rockchip-request@lists.infradead.org?subject=subscribe>
Sender: "Linux-rockchip" <linux-rockchip-bounces@lists.infradead.org>
Errors-To: 
 linux-rockchip-bounces+linux-rockchip=archiver.kernel.org@lists.infradead.org

Add a SPDIF audio-graph-card to ROCK Pi 4 device tree.

It's not enabled by default since all dma channels are used by
the (already) enabled i2s0/1/2 and the pin is muxed with GPIO4_C5
which might be in use already.
If enabled SPDIF_TX will be available at pin #15.

Signed-off-by: Alex Bee <knaerzche@gmail.com>
---
 .../boot/dts/rockchip/rk3399-rock-pi-4.dtsi   | 26 +++++++++++++++++++
 1 file changed, 26 insertions(+)

diff --git a/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4.dtsi b/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4.dtsi
index 80925a58e470..b5b8a79116fd 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4.dtsi
@@ -42,6 +42,23 @@ sound {
 		dais = <&i2s0_p0>;
 	};
 
+	sound-dit {
+		compatible = "audio-graph-card";
+		label = "SPDIF";
+		dais = <&spdif_p0>;
+	};
+
+	spdif-dit {
+		compatible = "linux,spdif-dit";
+		#sound-dai-cells = <0>;
+
+		port {
+			dit_p0_0: endpoint {
+				remote-endpoint = <&spdif_p0_0>;
+			};
+		};
+	};
+
 	vcc12v_dcin: dc-12v {
 		compatible = "regulator-fixed";
 		regulator-name = "vcc12v_dcin";
@@ -632,6 +649,15 @@ &sdhci {
 	status = "okay";
 };
 
+&spdif {
+
+	spdif_p0: port {
+		spdif_p0_0: endpoint {
+			remote-endpoint = <&dit_p0_0>;
+		};
+	};
+};
+
 &tcphy0 {
 	status = "okay";
 };
